02_bubble/nicholascross/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, bubble([]int{3, 2, 1, 5}))
+	fmt.Fprintln(out, insertion([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
+	count := len(s)
+	sort := make([]int, count)
+	copy(sort, s)
+
+	for sorted := false; !sorted; {
+		sorted = true
+
+		for i := 0; i+1 < count; i++ {
+			if sort[i] > sort[i+1] {
+				sort[i+1], sort[i] = sort[i], sort[i+1]
+				sorted = false
+			}
+		}
+	}
+
+	return sort
+}
+
+func insertion(s []int) []int {
+	count := len(s)
+	sort := make([]int, count)
+	copy(sort, s)
+
+	for i := 0; i < count; i++ {
```

Comment:
Since `i` is only used to initialise `j`, and the inner loop terminates immediately on `j == 0` (and hence `i == 0`), the whole `i == 0` case is a no-op, so this may as well start with `i := 1`.

As this is related to general programming and not Go-specific, not fixing this will not prevent approval or merging.